plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java

```diff
@@ -146,6 +154,28 @@ private static void removeParticipantsFromDiagram(List<InstanceRole> instanceRol
     }
     for (InstanceRole ir : irToRemove) {
       instanceRoles.remove(ir);
+      removeEditorMessages(messagesOrReferences, ir.getName());
+    }
+  }
+
+  /**
+   * Remove from xtext model all messages containing the participant that has just been deleted
+   * 
+   * @param messagesOrReferences
+   *          All messages and references from textual editor
+   * @param participantName
+   *          The participant that has just been deleted
+   */
+  private static void removeEditorMessages(EList<EObject> messagesOrReferences, String participantName) {
+    List<EObject> messages = messagesOrReferences.stream()
+        .filter(x -> x instanceof org.polarsys.capella.scenario.editor.dsl.textualScenario.SequenceMessage)
+        .collect(Collectors.toList());
```

Comment:
Since we do a for below to go trough the messages, no need to filter here, since it will introduce an overload. You can check in the for if is instanceof SequenceMessage and remove the filter part.